packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts

```diff
@@ -49,11 +49,26 @@ export interface Event {
  *
  * @experimental
  */
-export class EvaluateExpression implements sfn.IStepFunctionsTask {
-  constructor(private readonly props: EvaluateExpressionProps) {
+export class EvaluateExpression extends sfn.TaskStateBase {
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly evalFn: lambda.SingletonFunction;
+
+  constructor(scope: cdk.Construct, id: string, private readonly props: EvaluateExpressionProps) {
+    super(scope, id, props);
+
+    this.evalFn = createEvalFn(this.props.runtime || lambda.Runtime.NODEJS_10_X, this);
+
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: [this.evalFn.functionArn],
+        actions: ['lambda:InvokeFunction'],
+      }),
+    ];
   }
 
-  public bind(task: sfn.Task): sfn.StepFunctionsTaskConfig {
+  protected renderTask(): any {
```

Comment:
Does it make sense to make `renderTask` internal?

Reply:
It was something we considered (and prefer). 

The rationale behind making it protected:

Concrete classes will implement the abstract protected `renderTask` defined in the base. 
The base class uses it in the implementation of [`toStateJson`](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-stepfunctions/lib/states/task-base.ts#L135-L145). The `toStateJson` method is used when rendering any state in the sfn module. Although it's not a method that we would expect customers to call directly, `protected` is where we landed here.

Reply:
updated to internal in #8768